src/back/services/events.ts

```diff
@@ -0,0 +1,93 @@
+import crypto from 'crypto'
+
+import { ErrorService } from '../../services/ErrorService'
+import { EventWithAuthor } from '../../shared/types/events'
+import { EventType, ProposalCreatedEvent, UpdateCreatedEvent, VotedEvent } from '../../shared/types/events'
+import { getProfiles } from '../../utils/Catalyst'
+import { DclProfile } from '../../utils/Catalyst/types'
+import { ErrorCategory } from '../../utils/errorCategories'
+import EventModel from '../models/Event'
+
+export class EventsService {
+  static async getLatest(): Promise<EventWithAuthor[]> {
+    const latestEvents = await EventModel.getLatest()
+    const addresses = latestEvents.map((event) => event.address)
+
+    const profiles = await getProfiles(addresses)
+    const addressesToProfile: Record<string, DclProfile> = profiles.reduce((acc, profile) => {
+      acc[profile.address] = profile
+      return acc
+    }, {} as Record<string, DclProfile>)
+
+    const latestEventsWithAuthor: EventWithAuthor[] = []
+    for (const event of latestEvents) {
+      const { address } = event
+      latestEventsWithAuthor.push({
+        author: addressesToProfile[address].username || address,
+        avatar: addressesToProfile[address].avatar,
```

Comment:
We should return something even if `getProfiles` fails for some reason (using the address and a default avatar)

```
author: addressesToProfile[address]?.username || address,
avatar: addressesToProfile[address]?.avatar || DEFAULT_AVATAR,
```